openpype/hosts/nuke/api/lib_rendersettings.py

```diff
@@ -0,0 +1,111 @@
+
+from openpype.lib import Logger
+from openpype.settings import (
+    get_current_project_settings,
+    get_system_settings
+)
+
+
+class RenderFarmSettings:
+    """ Class for getting farm settings from project settings
+    """
+    log = Logger.get_logger("RenderFarmSettings")
+
+    _active_farm_module: str = None
+    _farm_modules: list = ["deadline"]
+    _farm_plugins: dict = {
+        "deadline": "NukeSubmitDeadline"
+    }
+    _creator_farm_keys: list = [
+        "chunk_size", "priority", "concurrent_tasks"]
+
+    _cached_project_settings = None
+    _cached_system_settings = None
+
+    def __init__(self, project_settings=None, log=None):
+        """ Get project settings and active farm module
+        """
+        if log:
+            self.log = log
+
+        if project_settings:
+            self._cached_project_settings = project_settings
+
+    @property
+    def project_settings(self):
+        """ returning cached project settings or getting new one
+        """
+        if not self._cached_project_settings:
+            self._cached_project_settings = get_current_project_settings()
+        return self._cached_project_settings
+
+    @property
+    def system_settings(self):
+        """ returning cached system settings or getting new one
+        """
+        if not self._cached_system_settings:
+            self._cached_system_settings = get_system_settings()
+        return self._cached_system_settings
+
+    def _get_active_farm_module_from_system_settings(self):
+        """ Get active farm module from system settings
+        """
+        active_modules = [
+            module_
+            for module_ in self._farm_modules
+            if self.system_settings["modules"][module_]["enabled"]
+        ]
```

Comment:
This should ideally use `ModulesManger` but if this is kept, it should use safe access to module keys.
```suggestion
        system_settings = self.system_settings
        active_modules = []
        for module_name in self._farm_modules:
            module_enabled = system_settings.get(module_name, {}).get("enabled")
            if module_enabled:
                active_modules .append(module_name
```